src/artemis_data_collector/artemis_data_collector.py

```
@@ -95,26 +99,47 @@ def run(self):
+        # Try primary URL first
         try:
             response = self.session.get(self.base_url + query)
+            if response.status_code == 200:
+                try:
+                    json_response = response.json()
+                    if json_response["status"] == 200:
+                        return json_response["value"]
+                    else:
+                        logger.error(f"Primary broker error: {json_response}")
+                except requests.exceptions.JSONDecodeError:
+                    logger.error(f"Primary broker JSON decode error: {response.text}")
```

Ruff (0.12.2)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)